File: 242_valid_anagram/242_valid_anagram.c
/*************************************************************************
	> File Name: 242_valid_anagram.c
	> Author: Tingjian Lau
	> Mail: tjliu@mail.ustc.edu.cn
	> Created Time: 2016/05/15
 ************************************************************************/

bool isAnagram(char* s, char* t) {
    int count[26] = {0};
    int len = strlen(s);
    int i;
        
    if(len != strlen(t)) return false;
        
    for(i = 0; i < len; i++){
        count[s[i] - 'a']++;
        count[t[i] - 'a']--;
    }
        
    for(i = 0; i < 26; i++) if(count[i]) 
        return false;
        
   return true;
}

File: 242_valid_anagram/242_valid_anagram.py
#!/usr/bin/env python
# -*- coding: UTF-8 -*-

##########################################################################
#	> File Name: 242_valid_anagram.py
#	> Author: Tingjian Lau
#	> Mail: tjliu@mail.ustc.edu.cn
#	> Created Time: 2016/05/15
#########################################################################

class Solution(object):
    def isAnagram_1(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        return sorted(s) == sorted(t)
        
    def isAnagram(self, s, t):
        if len(s) != len(t):
            return False
            
        dict = {}
        for item in s:
            dict[item] = dict.get(item, 0) + 1  # 如果找不到对应字符时，将其键值设为0，否则加一
        for item in t:
            dict[item] = dict.get(item, 0) - 1;
            
        for count in dict.values():
            if count is not 0:
                return False
        
        return True
            
        
